test/e2e/gpu_nodepools_create_delete.go

```diff
@@ -0,0 +1,151 @@
+	}
+
+	for _, sku := range gpuSkus {
+		sku := sku
+		It("creates and deletes vm type "+sku.display+" in a single cluster",
+			labels.RequireNothing,
+			labels.Critical,
+			labels.Positive,
+			func(ctx context.Context) {
+				const (
+					customerClusterName = "gpu-nodepool-cluster"
+					location            = "uksouth"
+				)
+
+				tc := framework.NewTestContext()
+
+				By("creating a resource group")
+				resourceGroup, err := tc.NewResourceGroup(ctx, "gpu-nodepools-"+sku.display, location)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("deploying demo template (single-step infra + identities + cluster)")
+				_, err = framework.CreateBicepTemplateAndWait(ctx,
+					tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+					*resourceGroup.Name,
+					"aro-hcp-demo",
+					framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/demo.json")),
+					map[string]interface{}{
+						"clusterName": customerClusterName,
+					},
+					45*time.Minute,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("getting credentials and verifying cluster is viable")
+				adminRESTConfig, err := framework.GetAdminRESTConfigForHCPCluster(
+					ctx,
+					tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient(),
+					*resourceGroup.Name,
+					customerClusterName,
+					10*time.Minute,
+				)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(framework.VerifyHCPCluster(ctx, adminRESTConfig)).To(Succeed())
+
+				// Use Bicep template to create a nodepool with the specified parameters
+				npName := "np-gpu-" + sku.display
+				By(fmt.Sprintf("creating GPU nodepool %q with VM size %q using Bicep template", npName, sku.vmSize))
+				_, err = framework.CreateBicepTemplateAndWait(ctx,
+					tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+					*resourceGroup.Name,
+					"aro-hcp-gpu-nodepool-"+sku.display,
+					framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/modules/nodepool.json")),
+					map[string]interface{}{
+						"clusterName":  customerClusterName,
+						"nodePoolName": npName,
+						"replicas":     1,
+						"vmSize":       sku.vmSize,
+					},
+					45*time.Minute,
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Verify provisioning succeeded and VM size matches what we requested
+				created, err := framework.GetNodePool(ctx,
+					tc.Get20240610ClientFactoryOrDie(ctx).NewNodePoolsClient(),
+					*resourceGroup.Name,
+					customerClusterName,
+					npName,
+					5*time.Minute,
+				)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(created.Properties).ToNot(BeNil())
+				Expect(created.Properties.ProvisioningState).ToNot(BeNil())
+				Expect(*created.Properties.ProvisioningState).To(Equal(hcpapi20240610.ProvisioningStateSucceeded))
+				Expect(created.Properties.Platform).ToNot(BeNil())
+				Expect(created.Properties.Platform.VMSize).ToNot(BeNil())
+				Expect(*created.Properties.Platform.VMSize).To(Equal(sku.vmSize))
+
```

Comment:
before we delete, check to ensure the new node is present using the `adminRESTConfig`